src/cli/command_info.rs

```diff
@@ -0,0 +1,174 @@
+use is_executable::IsExecutable;
+use itertools::Itertools;
+use std::env;
+use std::path::PathBuf;
+
+/// Find a specific external subcommand by name
+/// Based on cargo's find_external_subcommand function
+pub fn find_external_subcommand(cmd: &str) -> Option<PathBuf> {
+    let command_exe = format!("pixi-{}{}", cmd, env::consts::EXE_SUFFIX);
+    search_directories()
+        .iter()
+        .map(|dir| dir.join(&command_exe))
+        .find(|path| path.is_executable())
+}
+
+/// Execute an external subcommand
+pub fn execute_external_command(args: Vec<String>) -> miette::Result<()> {
+    if args.is_empty() {
+        return Err(miette::miette!("No command provided"));
+    }
+
+    let cmd = &args[0];
+    let cmd_args = &args[1..];
+
+    if let Some(path) = find_external_subcommand(cmd) {
+        // Execution
+        let mut command = std::process::Command::new(&path);
+        command.args(cmd_args);
+
+        // Set environment variables that extensions might need
+        if let Ok(current_dir) = env::current_dir() {
+            command.env("PIXI_PROJECT_ROOT", current_dir);
+        }
+
+        let status = command
+            .status()
+            .map_err(|e| miette::miette!("Failed to execute external command '{}': {}", cmd, e))?;
+
+        // Exit with the same code as the external command
+        if !status.success() {
+            if let Some(code) = status.code() {
+                std::process::exit(code);
+            } else {
+                std::process::exit(1);
+            }
+        }
+
+        Ok(())
+    } else {
+        // Command not found, try to find similar commands
+        let suggestions = find_similar_commands(cmd);
+        let help_message = if suggestions.is_empty() {
+            "help: view all installed commands with `pixi --list`".to_string()
+        } else {
+            format!(
+                "help: view all installed commands with `pixi --list`\n\nDid you mean '{}'?",
+                suggestions.join("', '")
+            )
+        };
+
+        Err(miette::miette!(
+            "No such command: `pixi {}`\n\n{}",
+            cmd,
+            help_message
+        ))
+    }
+}
+
+///fuzzy matching (built-in + external)
+fn find_similar_commands(cmd: &str) -> Vec<String> {
```

Comment:
let's remove the part of similar commands as we need another implementation, hardcode way is not extensible.

and we will keep this PR more focused